src/notifications/notificationmanager.cpp

```diff
@@ -212,7 +340,53 @@ uint NotificationManager::Notify(const QString &appName, uint replacesId, const
                                  const QString &summary, const QString &body, const QStringList &actions,
                                  const QVariantHash &hints, int expireTimeout)
 {
-    NOTIFICATIONS_DEBUG("NOTIFY:" << appName << replacesId << appIcon << summary << body << actions << hints << expireTimeout);
+    uint id = 0;
+    if (!calledFromDBus()) {
+        id = handleNotify(getpid(), appName, replacesId, appIcon, summary, body, actions, hints, expireTimeout);
+    } else {
+        setDelayedReply(true);
+        ClientIdentifier *identifier = new ClientIdentifier(this, connection(), message());
+        connect(identifier, &ClientIdentifier::finished, this, &NotificationManager::identifiedNotify, Qt::QueuedConnection);
+    }
+    return id;
+}
+
+void NotificationManager::identifiedNotify()
+{
+    ClientIdentifier *identifier = dynamic_cast<ClientIdentifier *>(sender());
```

Comment:
Does this `dynamic_cast` provide anything over `static_cast` if you don't do `nullptr` check afterwards? Although I guess referencing `nullptr` instead of "some random pointer" is safer.

Reply:
Basically after taking a refresh course on "those weird c++ casts", dynamic_cast just seemed to fit what is happening here -> I used it.

But now that you mentioned it, I'd consider speedy crash on null deref an improvement over limping onward after type confusion too ;-)

Reply:
There's also [qobject_cast](https://doc.qt.io/qt-5/qobject.html#qobject_cast) if you want even more choice 🙂

Reply:
I'll assume qt has a good reason for having such construct ("doesn't require RTTI support and it works [in situation where dynamic cast presumably would not]") -> switching to qobject_cast